third_party/txt/src/txt/run_metrics.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIB_TXT_SRC_RUN_METRICS_H_
+#define LIB_TXT_SRC_RUN_METRICS_H_
+
+#include "text_style.h"
+#include "third_party/skia/include/core/SkFontMetrics.h"
+
+namespace txt {
+
+// Contains the font metrics and TextStyle of a unique run.
+class RunMetrics {
+ public:
+  RunMetrics(const TextStyle* style) : text_style_(style) {}
+
+  RunMetrics(const TextStyle* style, SkFontMetrics& metrics)
+      : text_style_(style), font_metrics_(metrics) {}
+
+  SkFontMetrics& GetFontMetrics() { return font_metrics_; }
+
+  const TextStyle& GetTextStyle() const { return *text_style_; }
+
+ private:
+  const TextStyle* text_style_;
```

Comment:
Explicitly disable copy and assign instead of depending on the const-ness of this member (which may change in the future) to achieve the same.